### ISSUE TITLE
[BUG] "Don't show again" button on "Make sure that you are logged in" banner has no effect

### ISSUE THREAD

**Issue by Terrance** [IsThereAnyDeal/AugmentedSteam#2104]
### Expected Behavior

Dismissing this banner via _Don't show again_ should be remembered, and it shouldn't show again.

### Actual Behavior

The banner reappears on every page.

### Context

I'm constantly seeing the "Make sure that you are logged into steamcommunity.com" banner prompt within `steamcommunity.com` itself (not the store):

![image](https://github.com/user-attachments/assets/f0ba3b90-2265-41e8-b548-fc77c465ff3f)

I use Firefox with strong privacy features enabled, which I understand can cause interaction issues between Steam's domain (#818).  I've yet to figure out exactly what isn't working though -- the (!) icon claims two failed requests, but the console doesn't offer much context and there are no failed HTTP requests that I can see:

```
Augmented Steam v4.2.1  https://augmentedsteam.com/
    [Page.ts:92:16](moz-extension://c5a2ee31-d569-4155-a365-18407ba3ef5f/src/js/Content/Features/Page.ts)
Feature loading complete, 11 successfully loaded, 0 failed to load, 0 didn't load due to dependency errors
    [Context.ts:59:16](moz-extension://c5a2ee31-d569-4155-a365-18407ba3ef5f/src/js/Content/Modules/Context/Context.ts)
Error: LoginError: community
    [undefined](chrome://devtools/content/webconsole/undefined)
```

However, I've yet to see any obvious signs of breakage (presumably just one or more features not loading at all) so this hasn't really been an issue.  The banner is a minor nuisance though, as I can't seem to dismiss it in any permanent way.

### Steps to Reproduce the Problem

  1. Visit `steamcommunity.com` with a configuration that causes the banner
  2. Click _Don't show again_
  3. Reload or visit another page
  4. Banner shows again

### Specifications

  - Augmented Steam Version: 4.2.1
  - Browser Version: Firefox 134

**Comment by magnuspraeda** [comment]
I am experiencing this issue as well and I am also at a loss as to what exactly is failing to load and, similarly, cannot get the banner to go away between pages or reloads.

**Comment by Daniel-Chalil** [comment]
I've had this error before sporadically. I tried examining the console and found the two errors to be for "hn" and "LoginError", however I too was unable to find any root cause in the files to show the cause for it. 

I signed out of steam, deleted all cookies/local storage (for both store and community pages) and attempted to recreate the error to no avail. I also tried recreating the error in another session/browser with signing in into store only, community only and both concurrently however that didn't seem to recreate the bug either even after several days of trying.  

you can try resetting the cookies/storage and see if that fixes the issue, and in the meanwhile if anyone was able to find steps to reproduce this bug reliably, I would be interested in knowing to track this down further.

**Comment by candela97** [comment]
Quoting myself:
The community login error only affects features that make cross-origin requests to the community. The only ones left I believe are show applicable coupons and highlight games with available coupons/gifts etc., so you can disable them if you don't want to see this error, or make sure to visit any community page every 24 hours.

"Don't show again" not working was an issue before but it was quite some time ago and the extension has changed a lot since then. I'll have to check what's going on 😅

**Comment by Daniel-Chalil** [comment]
> Quoting myself: The community login error only affects features that make cross-origin requests to the community. The only ones left I believe are show applicable coupons and highlight games with available coupons/gifts etc., so you can disable them if you don't want to see this error, or make sure to visit any community page every 24 hours.
> 
> "Don't show again" not working was an issue before but it was quite some time ago and the extension has changed a lot since then. I'll have to check what's going on 😅

Thanks a lot for all the info because I'm finally seeing the error again. It seems that two things are happening here. 1) The handleClose function in LoginWarning.svelte is writing to localstorage to hide the login warning however I don't see any changes in local storage after hitting the hide button. 2) I don't see anywhere in AugmentedSteamWarnings or any other file the use of this data in local storage (hide_login_warn_store or hide_login_warn_community) to not dispatch the warning again if set to true.

**Comment by Aronjlove** [comment]
I encountered something similar, but disabling my ad-block for the Steam.com domain resolved the issue. I would see errors in my browser console about failing to connect to community but when I clicked the community tab I was logged in. After disabling uBlock Origin the errors stopped appearing, even after a reboot.

**Comment by Terrance** [comment]
To be clear, the scope of this issue was specifically for the "Don't show again" button of the banner not working as advertised, not that the banner is showing in the first place.

The appearance of the banner is a more general problem for which there are several past issues, one of which I linked as the likely culprit in my case.  That said, I'm now seeing the banner in the store as well as community without any changes to extension settings, so perhaps something has changed on Steam's end.

**Comment by Aronjlove** [comment]
> To be clear, the scope of this issue was specifically for the "Don't show again" button of the banner not working as advertised, not that the banner is showing in the first place.
> 
> The appearance of the banner is a more general problem for which there are several past issues, one of which I linked as the likely culprit in my case. That said, I'm now seeing the banner in the store as well as community without any changes to extension settings, so perhaps something has changed on Steam's end.

Yes, I assumed uBo on Edge was doing something similar as Firefox's strong privacy features, causing the button to not work for me. The banner goes away when I click on the community tab, but it returns when I go back to a game's store page.

**Comment by codetheron** [comment]
Maybe simply let us to disable this banner? :) I've deleted cookies, it helped ... for few days. Ad blocking disabled (in Vivaldi).

**Comment by Tenome** [comment]
I also have this, next time I see it I'll just uBlock it.

**Comment by codetheron** [comment]
> Same symptoms here. Thank you [@Ranshye](https://github.com/Ranshye)! Reconnecting IsThereAnyDeal fixed the banner and console "hn" error for me too.

Yes, somethings is going on out there.  Fixed my problem too on:
- Augmented Steam 4.2.1
- Vivaldi 7.1.3570.54 on Mac OS

Thanks :)
